#159965269 Implement articles CRUD

micahoriaso opened the pull request:
#### What does this PR do?
Implement articles crud

#### Description of tasks to be completed
- User should be able to create an article
- User should be able to update an article
- User should be able to delete an article
- User should be able to view all articles
- User should be able to view a single article

#### How should this be manually tested?
$ git clone https://github.com/andela/mag-ah-frontend.git
$ git checkout ft-articles-crud-159965269
$ cd mag-ah-frontend
$ npm install
$ npm start

#### What are the relevant pivotal tracker stories?
[#159965269](https://www.pivotaltracker.com/story/show/159965269t)

#### Screenshots
##### All articles
<img width="1135" alt="screen shot 2018-10-26 at 10 05 27" src="https://user-images.githubusercontent.com/20840601/47550076-b84ec000-d906-11e8-8400-bcedb62a7dcf.png">

##### Single Article
<img width="1210" alt="screen shot 2018-10-26 at 10 08 34" src="https://user-images.githubusercontent.com/20840601/47550184-24c9bf00-d907-11e8-9513-30ee863fc51e.png">

##### New Article
<img width="1215" alt="screen shot 2018-10-26 at 10 09 20" src="https://user-images.githubusercontent.com/20840601/47550240-44f97e00-d907-11e8-839c-b4f94054e7df.png">

##### Update Article
<img width="1229" alt="screen shot 2018-10-26 at 10 10 28" src="https://user-images.githubusercontent.com/20840601/47550281-7114ff00-d907-11e8-8972-95b7768e52eb.png">

##### My Articles also showing edit and delete
<img width="1289" alt="screen shot 2018-10-26 at 10 11 32" src="https://user-images.githubusercontent.com/20840601/47550321-94d84500-d907-11e8-8ffd-c5ddb5127878.png">

micahoriaso committed:
feat(articles): list all articles
- Create reducer for getting all articles
- Create actions for getting all articles
- Create Articles component

[WIP #159965269]

micahoriaso committed:
feat(articles-crud): create and list all articles

- List all articles
- Create an article

[WIP #159965269]

micahoriaso committed:
feat(articles): implement create article

- Implement create article

[WIP #159965269]

micahoriaso committed:
feat(articles): delete article

- Delete article actions
- Delete article reducer
- My articles component

[WIP #159965269]

micahoriaso committed:
feat(articles): delete article

- Delete article actions
- Delete article reducer
- My articles component
- Fix linting warnings

[WIP #159965269]

Bileonaire commented:
Looks good to me

kimotho-njoki commented on src/components/articles/MyArticles.jsx:
```diff
@@ -0,0 +1,81 @@
+
+class Articles extends Component {
+  constructor(props) {
+    super(props);
```
Is this constructor necessary?

micahoriaso replied:
Fixed

kimotho-njoki commented on src/components/articles/MyArticles.jsx:
```diff
@@ -0,0 +1,81 @@
+  };
+};
+
+export default connect(
```
Refactor to
export default connect(mapStateToProps, {
fetch: blah,
trash: blah,
reset: blah
})(Articles);

micahoriaso replied:
Fixed

kimotho-njoki commented on src/components/articles/UpdateArticle.jsx:
```diff
@@ -0,0 +1,151 @@
+                  value={body}
+                  onChange={this.handleChange}
+                  className={
+                    error.body
```
Nice job on the arrangement. Improvement.

micahoriaso replied:
Thank you!

mikenthiwa committed:
(feat articles-crud) Implement create, edit and delete article reducer test

kimotho-njoki commented on src/redux/action_types/index.js:
```diff
@@ -17,3 +17,17 @@ export const SOCIAL_LOGIN_START = "SOCIAL_LOGIN_START";
 export const SOCIAL_LOGIN_SUCCESS = "SOCIAL_LOGIN_SUCCESS";
 export const SOCIAL_LOGIN_FAIL = "SOCIAL_LOGIN_FAIL";
 export const RESET_STATE = "RESET_STATE";
+// Article action types
```
Unnecessary comment. Delete it. Remove the rest as well in this file. By looking at the constant I can understand what it is for.

micahoriaso replied:
Fixed

mikenthiwa committed:
(feat articles-crud) Implement create, edit and delete article reducer test

kimotho-njoki commented on src/redux/actions/fetchArticles.js:
```diff
@@ -0,0 +1,57 @@
+ *
+ * @return {object} type
+ */
+export const startArticleFetch = () => ({
```
Why import `startFetch()` in all the previous files but in this instance create an entire action creator?

micahoriaso replied:
Fixed

micahoriaso committed:
feat(articles): implement PR review feedback

- Fix header container width
- Remove unecessary constructor
- Fix code formatting
- Add test for components

[FINISHES #159965269]

micahoriaso committed:
Merge branch 'ft-articles-crud-159965269' of https://github.com/andela/mag-ah-frontend into ft-articles-crud-159965269

kimotho-njoki commented on src/redux/actions/viewSingleArticle.js:
```diff
@@ -0,0 +1,41 @@
+const baseUrl = config.BASE_URL;
+
+const viewSingleArticle = slug => async dispatch => {
+  /**
```
There should always be an empty line before a comment.

kimotho-njoki commented on src/redux/actions/viewSingleArticle.js:
```diff
@@ -0,0 +1,41 @@
+   * @param (object) article
+   * @return (object) type and payload
+   */
+
```
This extra line should not be here. No extra line after a comment.

kimotho-njoki commented on src/redux/actions/viewSingleArticle.js:
```diff
@@ -0,0 +1,41 @@
+
+const viewSingleArticle = slug => async dispatch => {
+  /**
+   * Successfull single article fetch
```
Is this how successful is spelled?
